aodn_cloud_optimised/bin/generic_cloud_optimised_creation.py

```diff
@@ -49,131 +27,236 @@
 from aodn_cloud_optimised.lib.s3Tools import s3_ls
 
 
+# ---------- Pydantic schema for validating the config ----------
+class ClusterConfig(BaseModel):
+    mode: Optional[str] = None
+    restart_every_path: bool = False
+
+
+class PathConfig(BaseModel):
+    s3_uri: str
+    filter: Optional[List[str]] = Field(default_factory=list)
+    year_range: Optional[List[int]] = None
```

Comment:
```suggestion
    year_range: Optional[List[int]] = None
    minimum_year: int | None = Field(default=None, description="The minimum year constraint, inclusive. None applies no constraint")
    maximum_year: int | None = Field(default=None, description="The maximum year constraint, inclusive. None applies no constraint")
```

I personally don't like using containers with implicit array indexes having a particular behaviour and prefer concrete properties.